File: Server/ServerFleetApi/ServerFleet.Models/Entities/Vehicle/VehicleModel.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ServerFleet.Models.Entities.Vehicle
{
    public class VehicleModel
    {
        public int Doors;

        public string Registration { get; set; }
        public string Make { get; set; }
        public string Model { get; set; }
        public string Colour { get; set; }
        public int Mileage { get; set; }
        public string Fuel { get; set; }
        public string BodyType { get; set; }
        public double Mpg { get; set; }
        public string Transmission { get; set; }
    }
}

File: Client/src/models/vehicle/VehicleModel.ts
import { IApiModel } from "../interfaces/IApiModel";

export default class VehicleModel  implements IApiModel {
    public static className = "VehicleModel"

    public vehicleId? : Number;
    public doors? : Number;
    public registration? : string;
    public make? : string;
    public model? : string;
    public colour? : string;
    public mileage? : Number;
    public fuel? : string;
    public bodyType? : string;
    public mpg? : Number;
    public transmission? : string;

    public get entityName(): string {
        return VehicleModel.className;
    }

    public get entityKey(): string {
        return `${this.vehicleId}`;
    }

    public get entityValue(): string {
        return `${this.registration}`;
    }

    public get entitySortValue(): any {
        return `${this.registration}`;
    }
}